src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -333,26 +319,28 @@ void reconnect() throws InterruptedException {
 
     void timeTrace(boolean trace, String format, Object... args) {
         if (trace) {
-            String timeStr = DateTimeFormatter.ISO_TIME.format(LocalDateTime.now());
-            System.out.printf("[%s] connect trace: ", timeStr);
-            System.out.printf(format, args);
-            System.out.println();
+            _trace(String.format(format, args));
+        }
+    }
 
+    void timeTrace(boolean trace, String message) {
+        if (trace) {
+            _trace(message);
         }
     }
 
-    long timeCheck(boolean trace, long endNanos, String format, Object... args) throws TimeoutException {
+    private void _trace(String message) {
+        String timeStr = DateTimeFormatter.ISO_TIME.format(LocalDateTime.now());
+        System.out.println("[" + timeStr + "] connect trace: " + message);
+    }

```

Comment:
I used underbar as in indicator for a private delegate. Other functions call (delegate) to this because they have common functionality, like the entire trace message code appeared twice.
In this case, I think it's reasonable for optimization. Not sure about printf, but print and printlin both are synchronized calls so you want to make as few as possible.